### PR TITLE
Changed attribute value creation route

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc. Feel free to add more information below this table.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See bug 2 on https://github.com/PrestaShop/PrestaShop/issues/35591
| UI Tests          | 
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Comment by nicosomb** [comment]
UI tests https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/8690654054

File: src/Core/Domain/AttributeGroup/ValueObject/AttributeGroupId.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public function getValue(): int
      */
     private function assertIntegerIsGreaterThanZero(int $attributeGroupId): void
     {
-        if (0 >= $attributeGroupId) {
+        if (0 > $attributeGroupId) {
```
Comment (jolelievre) [inline src/Core/Domain/AttributeGroup/ValueObject/AttributeGroupId.php]: The VO for IDs must validate that the value is valid, so strictly positive This shouldn't be modified

Why did you need to change this? Is it because of the form type that fetches the parent AttributeGroup? If so then it should be check in the form type itself not to fetch anything when `$attributeGoupId` is 0 (which it actually seems to be doing already)

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/attribute.yml [hunk 2]
```diff
@@ -26,16 +26,6 @@ admin_attributes_search:
     redirectQueryParamsToKeep:
       - 'attributeGroupId'
 
-admin_attributes_create:
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/attribute.yml:27]: I'm not sure this should be moved, the route is about Attribute, it's bound to AttributeController so it better fits in this file The fact that the legacy controller is `AdminAttributesGroups` is misleading but it makes sense because in the legacy page there was a unique controller to handle both values and groups

So they share the same parent legacy controller and the same permission but we can keep the routes apart

File: src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php [hunk 3]
```diff
@@ -100,14 +106,17 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-        if ($attributeGroup->group_type === AttributeGroupType::ATTRIBUTE_GROUP_TYPE_COLOR) {
-            $builder->add('color', ColorType::class, [
-                'label' => $this->trans('Color', 'Admin.Global'),
-                'required' => false,
-            ])->add('texture', FileType::class, [
-                'label' => $this->trans('Texture', 'Admin.Global'),
-                'required' => false,
-            ]);
+
+        if ($hasAttributeGroupId === true) {
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php]:
```suggestion
        if ($hasAttributeGroupId === true && $attributeGroup->group_type === AttributeGroupType::ATTRIBUTE_GROUP_TYPE_COLOR) {
```

Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php]: I'm not sure about the condition here, it makes sense to only add these fields when the parent is a color group BUT if we don't add it at all it means the fields are never present, it also means when you dynamically switch the attribute group in the page it can't toggle those fields

I'm not sure if the dynamic switch was handled in the page though but I guess it should since the legacy page handled this. So in this form type instead of removing the fields completely they should always be added, but their default visible state can be dynamized based on the attribute group type

Basically if the parent is not a group we could add a class `d-none` to hide the fields

**Review by jolelievre — CHANGES_REQUESTED** [review]

File: src/Core/Domain/AttributeGroup/ValueObject/AttributeGroupId.php (hunk 1 above)
Comment (nicosomb) [reply src/Core/Domain/AttributeGroup/ValueObject/AttributeGroupId.php]: Because it’s called « is greater » and not « isGreaterOrEqual ».

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/_catalog.yml [hunk 4]
```diff
@@ -27,13 +27,7 @@ _features:
   prefix: /features
 
 _attributes:
-  resource: "attribute.yml"
```
Comment (nicosomb) [inline src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/_catalog.yml:30]: As discussed with @jolelievre, we decided to merge `attribute.yml` and `attribute_group.yml`. Like that, we can handle routes with `{attributeGroupId}` parameter and those without it in the same file.

File: src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php (hunk 3 above)
Comment (nicosomb) [reply src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php]: This bug is a known one (it's the 3rd bug here https://github.com/PrestaShop/PrestaShop/issues/35591) and will be fixed in an other PR.

**Review by boherm — APPROVED** [review]

**Comment by nicosomb** [comment]
UI tests are green https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/8690654054

**Review by jolelievre — APPROVED** [review]
Thanks @nicosomb

**Review by florine2623 — APPROVED** [review]
Hello @nicosomb ,

LGTM ✅ 

Button to add a new value is well displayed on attributes listing page.

Thanks!
